tools/querytee/response_comparator.go

```diff
@@ -186,16 +191,24 @@ func compareScalar(expectedRaw, actualRaw json.RawMessage) error {
 	}, model.SamplePair{
 		Timestamp: actual.Timestamp,
 		Value:     actual.Value,
-	})
+	}, tolerance)
 }
 
-func compareSamplePair(expected, actual model.SamplePair) error {
+func compareSamplePair(expected, actual model.SamplePair, tolerance float64) error {
 	if expected.Timestamp != actual.Timestamp {
 		return fmt.Errorf("expected timestamp %v but got %v", expected.Timestamp, actual.Timestamp)
 	}
-	if expected.Value != actual.Value {
+	if !compareSampleValue(expected.Value, actual.Value, tolerance) {
 		return fmt.Errorf("expected value %s for timestamp %v but got %s", expected.Value, expected.Timestamp, actual.Value)
 	}
 
 	return nil
 }
+
+func compareSampleValue(first, second model.SampleValue, tolerance float64) bool {
+	if tolerance <= 0 {
+		return first == second
+	} else {
+		return math.Abs(float64(first)-float64(second)) <= tolerance
```

Comment:
Tolerance is generally a ratio or number of decimal places, not a absolute difference.

Reply:
Yeah. You might want to look at https://github.com/beorn7/floats/blob/master/equal.go (shameless plug).